synapse/rest/client/v1/login.py

```diff
@@ -336,10 +347,14 @@ async def _do_jwt_login(self, login_submission: JsonDict) -> Dict[str, str]:
                 errcode=Codes.FORBIDDEN,
             )
 
-        user = payload.get("sub", None)
+        subject_claim = self.jwt_subject_claim or "sub"
+        user = payload.get(subject_claim, None)
```

Comment:
We already did the fallback to `sub` in the config code, no need to do it again.

```suggestion
        user = payload.get(self.jwt_subject_claim, None)
```

Reply:
The issue is that tests don't use the config code but set `hs.config` directly, so without specifying a fallback here, many tests in `JWTTestCase` would fail. What do you suggest to solve this?

Reply:
The config code should get run during tests, see:

https://github.com/matrix-org/synapse/blob/fe604a022a7142157da7e90a40330beb2a11af7a/tests/unittest.py#L469-L472

Reply:
Were you able to try this again? It should run fine during tests.